### DISCUSSION TITLE
Does Anserini support indexing like COIL?

### DISCUSSION THREAD

**Discussion by namespace-Pt** [castorini/anserini#1716] [Q&A] [answered]
COIL saves each token's dense embedding and retrieves document based on the dot product between overlapping tokens. Can Anserini do this? Or would you implement this in the future?

**Comment by lintool** [comment, marked as answer]
Currently, Anserini does not support COIL and there are no plans to add support for COIL.

Anserini _does_ support uniCOIL (our improvement on COIL), as described here: https://arxiv.org/abs/2106.14807

See reproduction guides here:
+ https://github.com/castorini/anserini/blob/master/docs/experiments-msmarco-unicoil.md
+ https://github.com/castorini/anserini/blob/master/docs/experiments-msmarco-passage-unicoil-tilde-expansion.md
